### PR TITLE
WIP: [Feature]MultiModality: Text

### PR DESCRIPTION
**Pull request by SuX97**
+ Add text embedding dataset(called AuxiliaryFeatureDataset) and pipeline
+ Add text embedding preparation tools and k400 description classification result and fusion rgb result
+ Add self-supervised training (A re-implementation of this [paper](https://arxiv.org/pdf/2001.05691.pdf))

### CODE REVIEW

File: tools/youtube_crawler.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    while True:
+        try:
+            page = requests.get(url).text
+            return page
+        except SSLError:
+            print(f'Connection error {url}.')
+            continue
+
+
+def get_title(page):
```
Comment (kennymckormick) [inline tools/youtube_crawler.py:22]: Actually, you can use youtube-dl to get metadata like title or descriptions or subtitles. No need to write your own code.

Comment (SuX97) [reply tools/youtube_crawler.py:22]: Right, I just missed that `--get-title` args.

File: val_titles.txt [hunk 2]
```diff
@@ -0,0 +1,1800 @@
+zY_NOm7Qb8s LAGUNA PARK GUAYAQUIL ECUADOR SLACKLINE
```
Comment (dreamerlin) [inline val_titles.txt:1]: Is this file added by mistake?

Comment (SuX97) [reply val_titles.txt:1]: Yeah, will be removed later.

File: mmaction/datasets/pipelines/loading.py [hunk 3]
```diff
@@ -955,6 +956,89 @@ def __call__(self, results):
...
+class LoadFrameLevelAuxiliaryFeature(object):
+    """Load frame level aux-feature.
+        Required keys are "featurepath" and "frame_inds",
+        added or modified keys are "features", "sampled_inds", "input_shape".
+        Args:
+            # FIXME
+    """
+
+    def __init__(self, raw_feature_ext='.json', feature_name='text'):
+        valid_raw_feature_ext = ('.json')
```
Comment (dreamerlin) [inline mmaction/datasets/pipelines/loading.py:969]: ('.json', )

**Pull request closed without merging**